content/post/date_features.Rmd

```diff
@@ -0,0 +1,184 @@
+hours = 0:23
+hours_scaled = (2 * pi * hours) / 24
+hours_sin = sin(hours_scaled)
+hours_cos = cos(hours_scaled)
+n = 24
+cols = rainbow(n, s = 1, v = 1, start = 0, end = max(1, n - 1) / n, alpha = 1)
+par(mfrow = c(1, 2))
+plot(hours, type = "n", xlab = "Index", ylab = "Hours")
+text(hours, col = cols, labels = as.character(hours))
+plot(hours_sin, hours_cos, type = "n", xlab = "Sinus Transformation", ylab = "Cosines Transformation")
+text(hours_sin, hours_cos, col = cols, labels = as.character(hours))
+```
+
+`r mlr_pkg("mlr3pipelines")` provides the `PipeOpDateFeatures` pipeline which can be used to
+automatically engineer features based on `POSIXct` columns, including handling of cyclical features.
+
+## Bike Sharing
+The [Bike Sharing Dataset](https://archive.ics.uci.edu/ml/datasets/bike+sharing+dataset) contains
+the hourly count of rental bikes between years 2011 and 2012 in Capital bikeshare system with the
+corresponding weather and seasonal information:
+```{r}
+tmp <- tempfile()
+download.file("https://archive.ics.uci.edu/ml/machine-learning-databases/00275/Bike-Sharing-Dataset.zip", tmp)
+bikes = read.csv(unz(tmp, filename = "hour.csv"), as.is = TRUE)
+bikes$season = factor(bikes$season, labels = c("winter", "spring", "summer", "fall"))
+bikes$holiday = as.logical(bikes$holiday)
+bikes$workingday = as.logical(bikes$workingday)
+bikes$weathersit = as.factor(bikes$weathersit)
+str(bikes)
+```
+
+The original dataset does not contain a `POSIXct` column, but we can easily generate one based on
+the other variables available (note that as no information regarding minutes and seconds is
+available, we set them to `:00:00`):
+```{r}
+bikes$date = as.POSIXct(strptime(paste0(bikes$dteday, " ", bikes$hr, ":00:00"), tz = "GMT",
+  format = "%Y-%m-%d %H:%M:%S"))
+summary(bikes$date)
+```
+
+Using a random forest, we want to predict the total count of rental bikes `cnt`.
+
+## Baseline
+We construct a new regression task and create a vector of train and test indices:
+```{r}
+library(mlr3)
+library(mlr3learners)
+set.seed(2906)
+tsk = TaskRegr$new("bikes", backend = bikes, target = "cnt")
+train.idx = sample(seq_len(tsk$nrow), size = 0.7 * tsk$nrow)
+test.idx = setdiff(seq_len(tsk$nrow), train.idx)
+```
+
+This allows us to construct a train and test task:
+```{r}
+tsk_train = tsk$clone()$filter(train.idx)
+tsk_test = tsk$clone()$filter(test.idx)
+```
+
+To estimate the performance on future data, we will use `5-fold cross-validation`:
+```{r}
+cv5 = rsmp("cv", folds = 5)
+```
+
+As our baseline model, we use a random forest, `r ref("mlr_learners_regr.ranger", "ranger learner")`.
+For the baseline, we only use the original features that are sensible and drop `instant` (record
+index), `dteday` (year-month-day as a `character`, not usable), `date` (our new `POSIXct` variable
+which we will only use later), `casual` (count of casual users) and `registered` (count of
+registered users). Note that `casual` and `registered` adds up to `cnt`.
+```{r}
```

Comment:
1 sentence why you drop casual and registered.
Those variables will not be available during prediction